src/service.js

```diff
@@ -125,18 +127,23 @@ class Service {
       return this._find(params).then(page => page.data);
     }
 
-    return this._get(id);
+    return this._get(id, params);
   }
 
   create(data) {
     return this.Model.create(data).catch(errorHandler);
   }
 
   update(id, data, params) {
-    if(id === null) {
+    if (id === null) {
       return Promise.reject('Not replacing multiple records. Did you mean `patch`?');
     }
 
+    // Handle case where data might be a mongoose model
+    if (typeof data.toObject === 'function') {
```

Comment:
I accidentally committed code from the other PR but for some reason even though I cleaned up the commits the diff is still showing the old stuff.